ccdproc/image_collection.py

```diff
@@ -112,6 +113,19 @@ def __init__(self, location=None, keywords=None, info_file=None,
         if keywords:
             self.keywords = keywords
 
+    def __repr__(self):
+
+        str_repr = ("{0}(location='{1}', keywords={2}, ").format(
```

Comment:
Instead of positional parameters you can also use named parameters, for example:

```
kw = '*' if self._all_keywords else self.keywords[1:]
r = "{self.__class__.__name__}(location={self._location}, keywords={kw})".format(self=self, kw=kw)
```

That also allows to make the base-representation (before formatting) depend on `self._info_file` and `self._filenames` instead of adding these.